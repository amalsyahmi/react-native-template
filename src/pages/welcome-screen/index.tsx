import * as React from "react";
import {SafeAreaView, Text, View, Button, Alert} from 'react-native';
import { useNavigation } from "@react-navigation/native";


export const WelcomeScreen = () => {

  const navigation = useNavigation();

  return (
    <SafeAreaView>
      <View style={{paddingVertical: 100, paddingHorizontal: 30}}>
        <Text style={{alignSelf: 'center', fontSize: 25, color: '#777'}}>
          by Amal Syahmi
        </Text>
        {/* Spacer */}
        <View style={{width: 30, height: 30}}></View>
        <Button
          onPress={() => Alert.alert('Alert !', 'alert sample')}
          title="Alert"
          color="#841584"
          accessibilityLabel="Learn more about this purple button"
        />
        <View style={{width: 10, height: 10}}></View>
        <Button
          onPress={() => navigation.reset({
            routes: [
              {
                name: "WelcomeScreen",
                params: { screen: "WelcomeScreen" },
              },
            ],
          })}
          title="Navigate to page"
          color="#234489"
          accessibilityLabel="Learn more about this purple button"
        />
      </View>
    </SafeAreaView>
  );
};
