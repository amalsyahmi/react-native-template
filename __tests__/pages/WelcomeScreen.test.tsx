import 'react-native';
import React from 'react';
import { create } from 'react-test-renderer';

import {WelcomeScreen} from '../../src/pages/welcome-screen';

const mockedNavigation = jest.fn();

jest.mock('@react-navigation/core', () => {
  const actualNav = jest.requireActual('@react-navigation/core');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigation,
      dispatch: jest.fn(),
    }),
  };
});

describe('WelcomeScreen page', () => {
  beforeEach(() => {
    mockedNavigation.mockClear();
  });

  it('renders correctly', () => {
    const ui = create(<WelcomeScreen />).toJSON();
    expect(ui).toMatchSnapshot();
  });
});
