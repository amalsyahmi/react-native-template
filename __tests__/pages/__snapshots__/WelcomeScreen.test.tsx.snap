// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen page renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "paddingHorizontal": 30,
        "paddingVertical": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#777",
          "fontSize": 25,
        }
      }
    >
      by Amal Syahmi
    </Text>
    <View
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <View
      accessibilityLabel="Learn more about this purple button"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Alert
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 10,
          "width": 10,
        }
      }
    />
    <View
      accessibilityLabel="Learn more about this purple button"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#234489",
              },
            ]
          }
        >
          Navigate to page
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
